src/pkg/services/m365/api/users.go

```diff
@@ -245,244 +159,184 @@ func (c Users) GetAllIDsAndNames(ctx context.Context, errs *fault.Bus) (idname.C
 
-		delete(userInfo.ServicesEnabled, path.OneDriveService)
+		return userInfo, nil
 	}
 
-	mbxInfo, err := c.getMailboxSettings(ctx, userID)
+	mboxSettings, err := c.getMailboxSettings(ctx, userID)
 	if err != nil {
-		return nil, err
-	}
+		logger.CtxErr(ctx, err).Info("err getting user's mailbox settings")
 
-	userInfo.Mailbox = mbxInfo
-
-	// TODO: This tries to determine if the user has hit their mailbox
-	// limit by trying to fetch an item and seeing if we get the quota
-	// exceeded error. Ideally(if available) we should convert this to
-	// pull the user's usage via an api and compare if they have used
-	// up their quota.
-	if mfs != nil {
-		mf := mfs.GetValue()[0] // we will always have one
-		options := &users.ItemMailFoldersItemMessagesDeltaRequestBuilderGetRequestConfiguration{
-			QueryParameters: &users.ItemMailFoldersItemMessagesDeltaRequestBuilderGetQueryParameters{
-				Top: ptr.To[int32](1), // just one item is enough
-			},
+		if !graph.IsErrAccessDenied(err) {
+			return nil, graph.Wrap(ctx, err, "getting user's mailbox settings")
 		}
-		_, err = c.Stable.Client().
-			Users().
-			ByUserId(userID).
-			MailFolders().
-			ByMailFolderId(ptr.Val(mf.GetId())).
-			Messages().
-			Delta().
-			Get(ctx, options)
-
-		if err != nil && !graph.IsErrQuotaExceeded(err) {
+
+		mi.ErrGetMailBoxSetting = append(mi.ErrGetMailBoxSetting, clues.New("access denied"))
+	} else {
+		mi = parseMailboxSettings(mboxSettings, mi)
+	}
+
+	err = c.getFirstInboxMessage(ctx, userID, ptr.Val(inbx.GetId()))
+	if err != nil {
+		if !graph.IsErrQuotaExceeded(err) {
 			return nil, err
 		}
 
 		userInfo.Mailbox.QuotaExceeded = graph.IsErrQuotaExceeded(err)
 	}
 
+	userInfo.Mailbox = mi
+
 	return userInfo, nil
 }
 
-// TODO: remove when exchange api goes into this package
-func (c Users) GetMailFolders(
+func (c Users) getMailboxSettings(
 	ctx context.Context,
 	userID string,
-	options users.ItemMailFoldersRequestBuilderGetRequestConfiguration,
-) (models.MailFolderCollectionResponseable, error) {
-	mailFolders, err := c.Stable.Client().Users().ByUserId(userID).MailFolders().Get(ctx, &options)
+) (models.Userable, error) {
+	settings, err := users.
+		NewUserItemRequestBuilder(
+			fmt.Sprintf("https://graph.microsoft.com/v1.0/users/%s/mailboxSettings", userID),
+			c.Stable.Adapter(),
+		).
+		Get(ctx, nil)
 	if err != nil {
-		return nil, graph.Wrap(ctx, err, "getting MailFolders")
+		return nil, graph.Stack(ctx, err)
 	}
 
-	return mailFolders, nil
+	return settings, nil
 }
 
-// TODO: remove when drive api goes into this package
-func (c Users) GetDrives(ctx context.Context, userID string) (models.DriveCollectionResponseable, error) {
-	drives, err := c.Stable.Client().Users().ByUserId(userID).Drives().Get(ctx, nil)
+func (c Users) GetMailInbox(
+	ctx context.Context,
+	userID string,
+) (models.MailFolderable, error) {
+	inbox, err := c.Stable.
+		Client().
+		Users().
+		ByUserId(userID).
+		MailFolders().
+		ByMailFolderId("inbox").
```

Comment:
TIL: folder name can be passed as folder id

Reply:
not normally.  This is a `well known folder` case: https://learn.microsoft.com/en-us/graph/api/resources/mailfolder?view=graph-rest-1.0